app/services/image_filename_service.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class ImageFilenameService
+  MAX_LENGTH = 65
+
+  def initialize(revision)
+    @revision = revision
+  end
+
+  def call(suggested_name, mime_type)
+    filename = ActiveStorage::Filename.new(suggested_name)
+
+    base = filename.base.parameterize.slice 0...MAX_LENGTH
+    base = ensure_unique(base)
+
+    extension = Rack::Mime::MIME_TYPES.invert[mime_type]
+    "#{base}#{extension}"
+  end
+
+private
+
+  attr_reader :revision
+
+  def image_revisions
+    @image_revisions ||= revision.image_revisions
+  end
+
+  def ensure_unique(base)
+    potential_conflicts = revision.image_revisions.map(&:filename)
```

Comment:
I don't think so, as a revision should fully represent all files that it has. If we make a new revision with more/fewer files, the available names will change accordingly - because the content of the document updates alongside changes to the images, the two should always be compatible.